node/src/state.rs

```diff
@@ -94,3 +96,124 @@ impl State {
         Some(current_global_slot / SLOTS_PER_EPOCH)
     }
 }
+
+#[serde_with::serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum P2p {
+    Pending(P2pConfig),
+    Ready(P2pState),
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum P2pInitializationError {
+    #[error("p2p is already initialized")]
+    AlreadyInitialized,
+}
+
+#[macro_export]
+macro_rules! p2p_ready {
```

Comment:
Can't we have a function with the same logic? I'm worried about growing compilation times.

Reply:
This is mostly for sorting out places where an action need p2p but it is not initialized. It turned into `panic` with proper info easily. Also we should bail out if there's no p2p (`return;`). As to compilation time, such macro rules is a drop in the ocean.